### PR TITLE
Retroseq

### PR DESCRIPTION
**Pull request by J35P312**
### This PR fixes:

- Added Retroseq command module and test

### Review:
- [ ] Code review
- [ ] New code is executed and covered by tests
- [ ] Tests pass

### CODE REVIEW

File: t/retroseq_call.t [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    retroseq_bed_path => {
+        input           => q{input.bed},
+        expected_output => q{-input} . $SPACE . q{input.bed},
+    },
+    reference_fasta_path => {
+        input           => q{reference.fa},
+        expected_output => q{-ref} . $SPACE . q{reference.fa},
+    },
+    output_prefix => {
+        input           => q{HappyKangaroo},
```
Comment (henrikstranneheim) [inline t/retroseq_call.t]: 😄

File: t/retroseq_call.t [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    },
+
+);
+
+my %specific_argument = (
+    infile_path => {
+        input           => q{infile_path},
+        expected_output => q{-bam infile_path},
+    },
+    retroseq_bed_path => {
```
Comment (henrikstranneheim) [inline t/retroseq_call.t]: Same here order AB

File: t/retroseq_discover.t [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+      . $MIP::Program::Retroseq::VERSION
+      . $COMMA
+      . $SPACE . q{Perl}
+      . $SPACE
+      . $PERL_VERSION
+      . $SPACE
+      . $EXECUTABLE_NAME );
+
+## Base arguments
+my @function_base_commands = qw{retroseq.pl};
```
Comment (henrikstranneheim) [inline t/retroseq_discover.t]:
```suggestion
my @function_base_commands = qw{ retroseq.pl };
```

**Comment by J35P312** [comment]
THanks!
It is fun to be back! =P
I have had a look at your comments, they should all be fixed now!

File: lib/MIP/Program/Retroseq.pm [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+            store => \$filehandle,
+        },
+        infile_path => {
+            defined     => 1,
+            required    => 1,
+            store       => \$infile_path,
+            strict_type => 1,
+        },
+        outputfile_path => {
+            required    => 1,
```
Comment (jemten) [inline lib/MIP/Program/Retroseq.pm]:
```suggestion
        outputfile_path => {
            defined     => 1,
            required    => 1,
```

Comment (jemten) [reply lib/MIP/Program/Retroseq.pm]: Or maybe it doesn't need to be defined?

File: lib/MIP/Program/Retroseq.pm [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+            store       => \$outputfile_path,
+            strict_type => 1,
+        },
+        reference_fasta_path => {
+            required    => 1,
```
Comment (jemten) [inline lib/MIP/Program/Retroseq.pm]:
```suggestion
            defined     => 1,
            required    => 1,
```

File: lib/MIP/Program/Retroseq.pm [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+            strict_type => 1,
+        },
+        mobile_element_tsv_path => {
+            defined     => 1,
+            required    => 1,
+            store       => \$mobile_element_tsv_path,
+            strict_type => 1,
+        },
+        outputfile_path => {
+            required    => 1,
```
Comment (jemten) [inline lib/MIP/Program/Retroseq.pm]:
```suggestion
            defined     => 1,
            required    => 1,
```

**Comment by J35P312** [comment]
Howdy!
Hope you are doing well in the Austrian Alps! =P
I have now added "defined" to all of those, I see that is how it is done in tiddit and gatk =P

**Pull request merged**